### ISSUE TITLE
Very unforgiving parser

### ISSUE THREAD

**Issue by pkej** [duartealexf/sql-ddl-to-json-schema#19]
I'm trying to import a [dump from mysql workbench](https://gist.github.com/pkej/7f7eebb2927e46e53d893b6a2bad8585). The workbench outputs lines with comments "-- " which the parser chokes on.

Then the "SET @OLD" statements throw an error.

After the first table in my dataset, no other tables are created, I'm not sure why. It even goes through some data without giving any error, yet producing no file for the table.

**Comment by pkej** [comment]
It also doesn't like ` around names, like in my dump.

To import I have to do a remove all lines starting with -- , all lines starting with SET, and all ` in the file.

**Comment by pkej** [comment]
Furthermore:
```sql
UNIQUE INDEX `i18nStringID_UNIQUE` (`Name_i18nStringID` ASC),
```
Throws:
```
Unexpected K_ASC token: "ASC"
```

**Comment by duartealexf** [comment]
Hi @pkej thank you a lot for opening this issue. I can only improve the parser by having feedback like yours.

Also thanks for providing the SQL you tried to parse.

Whoa! I see what you mean by "unforgiving" 😝. You are right about most issues pointed out, although the parser does support backticks (`).

Anyways, from this issue I took your SQL and did some experimenting. I ended up creating 8 other issues (referenced above) that need to be solved so the parser correctly parses your SQL.

So, when I was experimenting, I did correctly parse your SQL and have the Compact JSON [in this gist](https://gist.github.com/duartealexf/79bb0023bcd08fa28309cfcba670227f). You can use it with this lib to convert to JSON Schema if you'd like.

I will start working on the new issues right away.

**Comment by pkej** [comment]
Thank you, I was happy to find your project, since there is a dearth of such converters for sql.

As for the foreign keys, I turned them off. The mysql Workbench has quite a few options for exporting the DDL. Since it is free, you could easily download it and test output from it. I do think it has the traditional Northwind example database, or else mysql has some test datasets as well, which can be imported into the workbench.

Thank you for this tool, I'm looking forward to continue using it, as my data model is just modelling the basics of our system for now, the business tables are the next part.

**Comment by pkej** [comment]
BTW, I was very pleased to see that you parse all the comments into the json schema. That's so satisfying.

**Comment by pkej** [comment]
Great, thanks for the gist, that's an early christmas. Yes, I noticed the backticks worked after all, I continued trying out stuff to probe the edges of the parser.

**Comment by duartealexf** [comment]
Thanks @pkej ... I know what you mean by not finding such converters in JS, that's exactly why I created this. Glad to know you'll keep using it.

Actually my final goal, is to create other projects that uses this parser to generate boilerplate code from SQL DDL to the desired NodeJS backend framework, with models, migrations, controllers, rules, etc. Maybe even some integration with [OpenAPI Specs](https://github.com/OAI/OpenAPI-Specification). But before that the parser needs to be 100%.

So far with this current version, you can try exporting the DDL with the following strategy: create tables first, then alter them adding the FKs later.

If you have any more issues or requests, let me know.

**Comment by duartealexf** [comment]
Hi @pkej 

I have just released v2.2.0, you should check it out 😏
All pointed issues have been solved and you can now parse your SQL ✅

Please do not hesitate to open more issues should you come across any problems.
